crypto/sm2/sm2_crypt.c

```diff
@@ -91,11 +91,15 @@ int sm2_ciphertext_size(const EC_KEY *key, const EVP_MD *digest, size_t msg_len,
 {
     const size_t field_size = ec_field_size(EC_KEY_get0_group(key));
     const int md_size = EVP_MD_size(digest);
+    size_t sz;
 
     if (field_size == 0 || md_size < 0)
         return 0;
 
-    *ct_size = 12 + 2 * field_size + (size_t)md_size + msg_len;
+    sz = 2 * ASN1_object_size(0, field_size + 1, V_ASN1_OCTET_STRING)
+         + ASN1_object_size(0, md_size, V_ASN1_INTEGER)
+         + ASN1_object_size(0, msg_len, V_ASN1_INTEGER);
+    *ct_size = ASN1_object_size(0, sz, V_ASN1_SEQUENCE);
```

Comment:
Unrelated modification.

Reply:
Oops, I was testing the code I pasted into another PR.  I'll remove.